tests/PHPUnit/Unit/DataAccess/LogQueryBuilder/JoinGeneratorTest.php

```diff
@@ -102,6 +102,34 @@ public function test_generate_getJoinString_manuallyJoinedAlready()
         $this->assertEquals($expected, $generator->getJoinString());
     }
 
+    public function test_generate_getJoinString_manuallyJoinedAlreadyWithCustomConditionInArray()
+    {
+        $generator = $this->generate(array(
+            'log_visit',
+            array('table' => 'log_conversion', 'joinOn' => 'log_visit.idvisit2 = log_conversion.idvisit2'),
+            'log_conversion'
+        ));
```

Comment:
FYI: Just pushed an update to instead throw an exception when this happens with detailed information where it happens and why. As solving it won't be trivial and is easier to instead just re-structure the array order for now if / when needed